### ISSUE TITLE
Incorporate in Joomla component

### ISSUE THREAD

**Issue by integr8d** [thomasjbradley/signature-pad#58]
Trying hard to implement this into a Joomla environment
I have moved all files into a signature-pad folder and adjusted paths to suit

in the default.php page i wish to use it i added

require_once 'signature-pad/lib/save-signature.php';
//signature pad add scripts to head
$document  =& JFactory::getDocument();
$document->addCustomTag('
<link rel="stylesheet" href="signature-pad/build/jquery.signaturepad.css">
   <!DOCTYPE html>
   <script src="signature-pad/build/jquery.signaturepad.min.js"></script>
   <script src="signature-pad/build/json2.min.js"></script>
  
  <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>');

and in position required
/\* signature module */  

```
 <?php
// A small trigger to flip between showing the signature form and the regeneration screen
if ($show_form) {
  require_once 'views/accept.php';
} else {
  require_once 'signature-pad/views/regenerate.php';
}
```

  ?>

<script src="signature-pad/build/jquery.signaturepad.min.js"></script>

<?php
    // Another trigger to write the appropriate Javascript to the page:
    // If we are showing the form, write the initialization code
    // If we are showing the final signature, write the regeneration code
    if ($show_form) :
  ?>

<script>
$(document).ready(function () {
$('.sigPad').signaturePad({drawOnly : true});
});
</script>

<?php else : ?>

<script>
$(document).ready(function () {
// Write out the complete signature from the database to Javascript
var sig = <?php echo $output; ?>;
$('.sigPad').signaturePad({displayOnly : true}).regenerate(sig);
});
</script>

<?php endif; ?>

<script src="signature-pad/build/json2.min.js"></script>
    

```
/*end of signature module */
```

have adjusted save signature with defined db variables
$dsn = 'mysql:dbname=mydbname_jos1;host=127.0.0.1';
    $user = 'mydbusername';
    $password = 'mydbpassword';

adjusted accept.php with path/file for php i am using
<form method="post" action="components/com_hoduma/views/detail/tmpl/default.php" class="sigPad">

built table in database: signatures and setup as per instructions
from phpmyadmin i can see only one entry and only id=5 and signature=name i typed in

  id    signator    signature   sig_hash    ip  created
   5                 test sig                     0

my folder structure

signature-pad

----build
    ------flashcanvas.js
    ------flashcanvas.swf
    ------journal.eot
    ------journal.svg
    ------journal.ttf
    ------journal.woff
    ------jquery.1.8.3.min.js (just in case)
    ------jquery.signaturepad.css
    ------jquery.signaturepad.min.js
    ------json2.min.js
    ------pen.cur
----lib  
    -----save-signature.php
----views
    ------accept.php
    ------regenerate.php
build.js
changelog.md
error_log
index.php
jquery.signaturepad.js
new-bsd-license
pen.psd
readme.md
version.tx

my error log
[13-Jan-2013 12:11:55 UTC] PHP Warning:  require_once(signature-pad/lib/save-signature.php) [<a href='function.require-once'>function.require-once</a>]: failed to open stream: No such file or directory in /home/myservername/public_html/workorder/signature-pad/index.php on line 5
[13-Jan-2013 12:11:55 UTC] PHP Fatal error:  require_once() [<a href='function.require'>function.require</a>]: Failed opening required 'signature-pad/lib/save-signature.php' (include_path='.:/usr/lib/php:/usr/local/lib/php') in /home/myservername/public_html/workorder/signature-pad/index.php on line 5
[13-Jan-2013 12:13:52 UTC] PHP Warning:  require_once(/signature-pad/lib/save-signature.php) [<a href='function.require-once'>function.require-once</a>]: failed to open stream: No such file or directory in /home/myservername/public_html/workorder/signature-pad/index.php on line 5
[13-Jan-2013 12:13:52 UTC] PHP Fatal error:  require_once() [<a href='function.require'>function.require</a>]: Failed opening required '/signature-pad/lib/save-signature.php' (include_path='.:/usr/lib/php:/usr/local/lib/php') in /home/myservernames/public_html/workorder/signature-pad/index.php on line 5

but think they were all before i found the form method in accept.php, cant produce any more entries on error log.

for some reason i cant put up screenshots, will try in second item

from firebug get this
 /\* signature module */

<p class="error">Please sign the document</p>

<p class="error">Please enter your name</p>

<label class="error" for="name">Print your name</label>
<input id="name" class="name error" type="text" value="" name="name">

<p class="drawItDesc">Draw your signature</p>

<ul class="sigNav">
<li class="drawIt">
<a href="#draw-it">Draw It</a>
</li>
<li class="clearButton">
<a href="#clear">Clear</a>
</li>
</ul>

<div class="sig sigWrapper">
<div class="typed"></div>
<canvas class="pad" height="55" width="198"></canvas>
<input class="output" type="hidden" name="output">
</div>

<button type="submit">I accept the terms of this agreement.</button>
<script src="/workorder/signature-pad/build/jquery.signaturepad.min.js">
<script>
<script src="/workorder/signature-pad/build/json2.min.js">
/*end of signature module */ 

visually its
full width div, red bg white text Please sign your document
2nd full width div, red gb white text Please enter your name
3rd full width div first part Print your name then textarea box then button I accept...

Any more suggestions?? happy to give access if necessary

Thanks

Sean
